src/linear_algebra/determinant.lean

```diff
@@ -319,4 +330,209 @@ begin
     exact hkx }
 end
 
+lemma upper_two_block_triangular_det (A  : matrix m m R) (B  : matrix m n R) (C  : matrix n m R)
+  (D  : matrix n n R) (hz : C = 0) :
+  (matrix.from_blocks A B C D).det = A.det * D.det :=
```

Comment:
Any reason for stating it this way and not `matrix.from_blocks A B 0 D` without hz?

Reply:
Nope, just missing the forest for the trees.